scripts/add-celltype-fields-scpca-meta.py

```diff
@@ -0,0 +1,157 @@
+library_df = pandas.read_csv(args.library_file, sep="\t", keep_default_na=False)
+
+# Read in project celltype file
+project_df = pandas.read_csv(
+    args.project_celltype_file, sep="\t", keep_default_na=False
+)
+
+
+# join library and project cell type metadata
+all_metadata_df = library_df.join(
+    project_df.set_index("scpca_project_id"), on="scpca_project_id", how="left"
+)
+# need to filter to only projects that have at least one reference (either SingleR or CellAssign)
+all_metadata_df = all_metadata_df[
+    all_metadata_df[["singler_ref_file", "cellassign_ref_file"]].notnull().all(1)
+]
+
+# remove any extra / at the end
+bucket = args.bucket.strip("/")
+checkpoints_prefix = args.checkpoints_prefix.strip("/")
+
+# set up S3
+s3 = boto3.resource("s3")
+s3_bucket = s3.Bucket(bucket)
+
+# go through every run id and modify scpca-meta.json files if present
+for run in all_metadata_df.itertuples():
+    print(f"Processing {run.scpca_run_id}")
+
+    # build paths to singleR and CellAssign checkpoints
+    celltype_checkpoints = {
+        "singler_checkpoint": f"{checkpoints_prefix}/celltype/{run.scpca_library_id}/{run.scpca_library_id}_singler/scpca-meta.json",
+        "cellassign_checkpoint": f"{checkpoints_prefix}/celltype/{run.scpca_library_id}/{run.scpca_library_id}_cellassign/scpca-meta.json",
+    }
+
+    # cell type fields that need to be updated
+    new_fields = {
+        "singler_model_file": f"s3://scpca-references/celltype/singler_models/{run.singler_ref_file}",
+        "cellassign_reference_file": f"s3://scpca-references/celltype/cellassign_references/{run.cellassign_ref_file}",
+    }
+
+    # for each checkpoint file, add cellassign and singler file path
+    for checkpoint_file in celltype_checkpoints.values():
+        # make sure file exists, otherwise just skip
+        try:
+            checkpoint_obj = (
+                s3_bucket.Object(checkpoint_file).get()["Body"].read().decode("utf-8")
+            )
+            checkpoint_meta = json.loads(checkpoint_obj)
+        except s3.meta.client.exceptions.NoSuchKey:
+            print(f"No cell type scpca-meta.json file for {run.scpca_run_id}")
+            continue
+
+        # only modify if existing fields don't match what's in the project metadata
+        if (
+            checkpoint_meta["singler_model_file"] == new_fields["singler_model_file"]
+            and checkpoint_meta["cellassign_reference_file"]
+            == new_fields["cellassign_reference_file"]
+        ):
+            print(
+                f"All fields are present, no updates for {run.scpca_run_id} to {checkpoint_file}"
+            )
+        else:
+            # find if versions are in the existing checkpoints
+            singler_version_present = re.search(
+                r"(\d{1}-\d{2}-\d{1}_model.rds$)", checkpoint_meta["singler_model_file"]
+            )
+            cellassign_version_present = re.search(
+                r"(20\d{2}-[01]\d{1}-\d{2}.tsv$)",
```

Comment:
Don't really need the `{1}` as long as I am at it...
```suggestion
                r"(20\d{2}-[01]\d-\d{2}.tsv$)",
```